src/x/net/http/errors_test.go

```diff
@@ -33,35 +34,64 @@ import (
 )
 
 func TestErrorRewrite(t *testing.T) {
+	invalidParamsRewriteFn := func(err error) error {
+		if xerrors.IsInvalidParams(err) {
+			return errors.New("rewritten error")
+		}
+		return err
+	}
+
+	noRewriteFn := func(err error) error {
+		return err
+	}
+
 	tests := []struct {
 		name           string
 		err            error
 		expectedStatus int
 		expectedBody   string
+		rewriteFn      ErrorRewriteFn
 	}{
 		{
 			name:           "error that should not be rewritten",
 			err:            errors.New("random error"),
 			expectedStatus: 500,
 			expectedBody:   `{"status":"error","error":"random error"}`,
+			rewriteFn:      invalidParamsRewriteFn,
 		},
 		{
 			name:           "error that should be rewritten",
 			err:            xerrors.NewInvalidParamsError(errors.New("to be rewritten")),
 			expectedStatus: 500,
 			expectedBody:   `{"status":"error","error":"rewritten error"}`,
+			rewriteFn:      invalidParamsRewriteFn,
+		},
+		{
+			name:           "error should have correct status",
+			err:            xerrors.NewInvalidParamsError(errors.New("bad param")),
+			expectedStatus: 400,
+			expectedBody:   `{"status":"error","error":"bad param"}`,
+			rewriteFn:      noRewriteFn,
+		},
+		{
+			name:           "deadline exceeded map to correct status",
+			err:            context.DeadlineExceeded,
+			expectedStatus: 504,
+			expectedBody:   `{"status":"error","error":"context deadline exceeded"}`,
+			rewriteFn:      noRewriteFn,
+		},
+		{
+			name:           "canceled mapped to correct status",
+			err:            context.Canceled,
+			expectedStatus: 400,
+			expectedBody:   `{"status":"error","error":"context canceled"}`,
+			rewriteFn:      noRewriteFn,
```

Comment:
I think it would be better to place these in a separate test (e.g. `func TestWriteErrorResponse()`) instead of placing them inside `TestErrorRewrite()`.

Reply:
done